feat: generic oidc provider using auth-identity

edpaget opened the pull request:
### Description

Add OSS infrastructure for OIDC-based authentication providers:

- OIDC discovery: automatic endpoint discovery from issuer URL
- Token handling: ID token validation, parsing, and claim extraction
- State management: encrypted state cookies for CSRF protection
- Schema validation: Malli schemas for OIDC configuration and tokens
- Common utilities: shared HTTP and request utilities

This provides a foundation for enterprise SSO integrations that use
OIDC/OAuth2 flows (e.g., Slack Connect, generic OIDC providers).

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

edpaget committed:
feat: add generic OIDC provider infrastructure

Add OSS infrastructure for OIDC-based authentication providers:

- OIDC discovery: automatic endpoint discovery from issuer URL
- Token handling: ID token validation, parsing, and claim extraction
- State management: encrypted state cookies for CSRF protection
- Schema validation: Malli schemas for OIDC configuration and tokens
- Common utilities: shared HTTP and request utilities

This provides a foundation for enterprise SSO integrations that use
OIDC/OAuth2 flows (e.g., Slack Connect, generic OIDC providers).

johnswanson commented on src/metabase/sso/oidc/common.clj:
```diff
@@ -0,0 +1,151 @@
+  []
+  (-> (nonce/random-bytes 32)
+      codecs/bytes->hex
+      (str/replace #"[+/=]" "")))
```
similar dead code, I think?

johnswanson commented on src/metabase/util/http.clj:
```diff
@@ -1,13 +1,21 @@
 
 (set! *warn-on-reflection* true)
 
+(defn- lower-case-en
```
Why not use `u/lower-case-en` here?

johnswanson reviewed (approved):
A couple small nits but lgtm

edpaget committed:
fixup! dead code clean up

There was a bunch of speculative code we didn't end up using when this
change from a 'configurable OIDC' to a just support slack course

edpaget committed:
fixup! ttls for caches for jwks/discovery

fixup! check for open redirect when creating oidc state

edpaget committed:
fixup! use valid-hosts? to check before making jwks/discovery requests

edpaget committed:
fixup! don't use hardcoded jwt algo

edpaget committed:
fixup! use java-time

edpaget committed:
fixup! kondo

edpaget committed:
fixup! use system.core not system.settings
